### PR TITLE
Update Language Server Protocol version to 3.17

### PR DESCRIPTION
**Pull request by paracycle**
Since 3.16, the LSP spec has changed how it stores and publishes the spec. That meant that the `generateFiles.ts` script was no longer working and needed extensive work to bring up-to-date.

Fortunately, the spec is now being published as a [JSON meta model file](https://github.com/microsoft/language-server-protocol/blob/7e7971fca4ceeab1078374e2d331c5d21278da99/_specifications/lsp/3.17/metaModel/metaModel.json), which allows this library to drop the TS dependency and to start processing the spec using pure Ruby.

Thus, this PR, drops all Node/JS/TS dependencies and introduces a new `generate_files.rb` script to do file generation. Using that script, the PR also updates the gem to version 3.17 of the protocol spec.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Change log
 
+## 3.16.0.0
```
Comment (mtsmfm) [inline CHANGELOG.md:3]:
```suggestion
## 3.17.0.0
```

Comment (paracycle) [reply CHANGELOG.md:3]: 🤦

**Comment by mtsmfm** [comment]
Some tests were failed because of my lack of maintenance, you don't have to care about that. I'll fix them in another PR.

**Review by mtsmfm — APPROVED** [review]

**Comment by mtsmfm** [comment]
Awesome!

**Comment by mtsmfm** [comment]
JFYI: I didn't think this change affects benchmark though, I checked benchmark just in case and the result is same-ish.

https://github.com/mtsmfm/language_server-protocol-ruby/pull/38#issuecomment-1229096084

https://github.com/mtsmfm/language_server-protocol-ruby/pull/34#issuecomment-907765126

**Comment by mtsmfm** [comment]
I've just released 3.17.0.0. https://rubygems.org/gems/language_server-protocol/versions/3.17.0.0

**Pull request merged**